Transports/com.community.netcode.transport.facepunch/Runtime/FacepunchTransport.cs

```diff
@@ -203,11 +210,19 @@ void IConnectionManager.OnDisconnected(ConnectionInfo info)
                 Debug.Log($"[{nameof(FacepunchTransport)}] - Disconnected Steam user {info.Identity.SteamId}.");
         }
 
-        void IConnectionManager.OnMessage(IntPtr data, int size, long messageNum, long recvTime, int channel)
+        unsafe void IConnectionManager.OnMessage(IntPtr data, int size, long messageNum, long recvTime, int channel)
         {
-            byte[] payload = new byte[size];
-            Marshal.Copy(data, payload, 0, size);
-            InvokeOnTransportEvent(NetworkEvent.Data, ServerClientId, new ArraySegment<byte>(payload, 0, size), Time.realtimeSinceStartup);
+            if (payloadCache.Length < size)
+            {
+                payloadCache = new byte[payloadCache.Length * 2];
+            }
+
+            fixed (byte* payload = payloadCache)
+            {
+                UnsafeUtility.MemCpy(payload, (byte*)data, size);
+            }
```

Comment:
It is safe to share the `payloadCache` since this method is not called from multiple threads. Additionally, the `NetworkManager` does not keep any reference to the data after `InvokeOnTransportEvent` returns.